internal/storer/storer.go

```diff
@@ -0,0 +1,131 @@
+// Package storer is implements a cache deleting its entries after each interval.
+package storer
+
+import (
+	"errors"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/newrelic/infra-integrations-sdk/persist"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// DefaultTTL is default ttl of the ache entries.
+	DefaultTTL = 10 * time.Minute
+	// DefaultInterval is default interval to execute the "garbage collection" of the cache.
+	DefaultInterval = 15 * time.Minute
+)
+
+// InMemoryStore is similar to the sdk one, the main difference is cleanCache method executed each interval.
+type InMemoryStore struct {
+	cachedData  map[string]jsonEntry
+	locker      *sync.RWMutex
+	ttl         time.Duration
+	logger      *logrus.Logger
+	ticker      *time.Ticker
+	stopChannel chan struct{}
+}
+
+// Holder for any entry in the JSON storage.
+type jsonEntry struct {
+	//notice this is the timestamp of the creation of the jsonEntry, we do not keep track of the last-access timestamp.
+	timestamp time.Time
+	value     interface{}
+}
+
+// NewInMemoryStore will create and initialize an InMemoryStore.
+func NewInMemoryStore(ttl time.Duration, interval time.Duration, logger *logrus.Logger) *InMemoryStore {
```

Comment:
new packages are using the option or/and config pattern , would make sense to implement it here as well?

Reply:
I wanted to keep the implementation as similar as possible to the sdk one, it is basically a copy paste with the cleaning.

Reply:
but it could make sense to update that to be similar and have better structure